Allow empty tokenEndpointUri in OAuth PLAIN to assume token in password

elakito opened the pull request:
### Type of change

_Select the type of your PR_
- Enhancement / new feature

### Description

_Please describe your pull request_

When the oauth plain mode is updated with https://github.com/strimzi/strimzi-kafka-oauth/pull/107, this change in strimzi-kafka-operator will allow this configuration by not forbidding an empty `TokenEndpointUri` value.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

scholzj commented on cluster-operator/src/test/java/io/strimzi/operator/cluster/model/ListenersValidatorTest.java:
```diff
@@ -748,7 +749,7 @@ public void testMinimalConfiguration() {
     public void testValidateOauthPlain() {
         KafkaListenerAuthenticationOAuthBuilder authBuilder = new KafkaListenerAuthenticationOAuthBuilder()
-                .withEnableOauthBearer(false);
+                .withEnableOauthBearer(false).withEnablePlain(true).withJwksEndpointUri("http://localhost:8080/jwks").withCheckIssuer(false);
```
I think this test test something else from what you are changing. If I read it right, it tests that you cannot disable SASL OAUTHBEARER without adding SASL PLAIN. So IMHO you should keep the original test and add your test as additional one.

scholzj commented:
Also, can you please fix the DCO signoff? The details how to do it are under the _Details_ link next to the DCO status.

elakito commented:
@scholzj OK. I'll update this PR.

elakito commented:
@scholzj I added the sing-off and updated the test so that the original test that checked the error "... 'tokenEndpointUri' has to be specified." for an empty tokenEndpointUri to check other errors and added a new test checks no "... 'tokenEndpointUri' has to be specified."  is thrown when tokenEndpointUri is not set.

mstruk reviewed (approved):
Looks good. We'll have to make some changes in the docs as well, but I can add another PR for that.

scholzj commented:
Note: we have to merge and release the OAuth library first before we can merge this.

scholzj commented:
@elakito Just FYI ... I'm working on the OAuth 0.7.2 release with this feature. So I rebased and updated the PR to use it from staging so that we can run tests against it and get it merged eventually.

scholzj commented:
/azp run regression

scholzj commented:
/azp run regression

scholzj commented:
Thanks for the PR @elakito
